app/src/main/java/org/akvo/flow/activity/RecordActivity.java

```diff
@@ -114,22 +121,39 @@ public void onSurveyClick(final String formId) {
             Toast.makeText(this, R.string.pleasewaitforbootstrap, Toast.LENGTH_LONG).show();
             return;
         }
-        Survey survey = mDatabase.getSurvey(formId);
+        final Survey survey = mDatabase.getSurvey(formId);
         if (!survey.isHelpDownloaded()) {
             Toast.makeText(this, R.string.error_missing_cascade, Toast.LENGTH_LONG).show();
             return;
         }
 
+        getSelectedUser.execute(new DefaultObserver<User>() {
+            @Override
+            public void onError(Throwable e) {
+                Timber.e(e);
+            }
+
+            @Override
+            public void onNext(User user) {
+                String userName = user.getName();
+                if (!TextUtils.isEmpty(userName)) {
+                    displayForm(user, formId, survey);
+                }
+            }
+        }, null);
+
+
+    }
+
+    private void displayForm(User user, String formId, Survey survey) {
         // Check if there are saved (non-submitted) responses for this Survey, and take the 1st one
-        long[] instances = mDatabase.getFormInstances(recordId, formId,
-                SurveyInstanceStatus.SAVED);
+        long[] instances = mDatabase.getFormInstances(recordId, formId, SurveyInstanceStatus.SAVED);
```

Comment:
I guess these are automatic changes by the IDE.  remember `git add -p` to exclude unless needed.

Reply:
I think it makes sense to include this change in here as creating a specific PR for just formatting seems an overhead for me and since I have changed this file it makes sense to include it here as well.